examples/nested_parallel_memory.py

```diff
@@ -0,0 +1,124 @@
+"""
+==================================================
+Checkpoint using joblib.Memory and joblib.Parallel
+==================================================
+
+We illustrate how to cache intermediate computing results using
+:class:`joblib.Memory` within :class:`joblib.Parallel`.
+
+"""
+
+###############################################################################
+# Embed caching within parallel processing
+###############################################################################
+# 
+# It is possible to cache a computationally expensive function executing during
+# a parallel process. ``costly_column`` will emulate such function.
+
+import time
+
+
+def costly_column(data, column):
+    """Emulate a costly function by sleeping and returning a column."""
+    time.sleep(2)
+    return data[column]
+
+
+def data_processing_mean(data, column):
+    """Compute the mean of a column."""
+    return costly_column(data, column).mean()
+
+
+###############################################################################
+# Create some data
+
+import numpy as np
+data = np.random.randn(int(1e4), 4)
+
+###############################################################################
+# It is first possible to make the processing without caching or parallel
+# processing.
+
+start = time.time()
+results = [data_processing_mean(data, col) for col in range(data.shape[1])]
+stop = time.time()
+
+print('\nSequential processing')
+print('Elapsed time for the entire processing: {:.2f} s'
+      .format(stop - start))
+
+###############################################################################
+# ``costly_column`` is expensive to compute and it is used as an intermediate
+# step in ``data_processing_mean``. Therefore, it is interesting to store the
+# intermediate results from ``costly_column`` using :class:`joblib.Memory`.
+
+from joblib import Memory
+
+cachedir = './cachedir'
+memory = Memory(cachedir=cachedir, verbose=0)
```

Comment:
I think that in the dev version, the preferred way of instanciating a Memory is by specifying "location", and not "cachedir".

Reply:
indeed, so the memory documentation should be updated...